comp/core/workloadmeta/collectors/internal/ecsfargate/v2parser.go

```diff
@@ -0,0 +1,173 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker
+
+// Package ecsfargate implements the ECS Fargate Workloadmeta collector.
+package ecsfargate
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	v2 "github.com/DataDog/datadog-agent/pkg/util/ecs/metadata/v2"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func (c *collector) parseTaskFromV2Endpoint(ctx context.Context) ([]workloadmeta.CollectorEvent, error) {
+	task, err := c.metaV2.GetTask(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return c.parseV2Task(task), nil
+}
+
+func (c *collector) parseV2Task(task *v2.Task) []workloadmeta.CollectorEvent {
+	events := []workloadmeta.CollectorEvent{}
+	seen := make(map[workloadmeta.EntityID]struct{})
+
+	// We only want to collect tasks without a STOPPED status.
+	if task.KnownStatus == workloadmeta.ECSTaskKnownStatusStopped {
+		return events
+	}
+
+	arnParts := strings.Split(task.TaskARN, "/")
+	taskID := arnParts[len(arnParts)-1]
+	entityID := workloadmeta.EntityID{
+		Kind: workloadmeta.KindECSTask,
+		ID:   task.TaskARN,
+	}
+
+	seen[entityID] = struct{}{}
+
+	taskContainers, containerEvents := c.parseTaskContainers(task, seen)
+	entity := &workloadmeta.ECSTask{
+		EntityID: entityID,
+		EntityMeta: workloadmeta.EntityMeta{
+			Name: taskID,
+		},
+		ClusterName: parseClusterName(task.ClusterName),
+		Region:      parseRegion(task.ClusterName),
+		Family:      task.Family,
+		Version:     task.Version,
+		LaunchType:  workloadmeta.ECSLaunchTypeFargate,
+		Containers:  taskContainers,
+
+		// the AvailabilityZone metadata is only available for
+		// Fargate tasks using platform version 1.4 or later
+		AvailabilityZone: task.AvailabilityZone,
```

Comment:
What happens here if we are using a platform with version older than 1.4? Does this return an empty string in that case?

Reply:
It would be empty as it's our current behaviour https://github.com/DataDog/datadog-agent/blob/main/comp/core/workloadmeta/collectors/internal/ecsfargate/ecsfargate.go#L122